squirrel.api/src/main/java/org/dice_research/squirrel/sink/tripleBased/AdvancedTripleBasedSink.java

```diff
@@ -18,4 +18,26 @@
      * @return All {@link Triple}s behind the given uri.
      */
     List<Triple> getTriplesForGraph(CrawleableUri uri);
+
+    /**
+     * This method drops the graph of the uri
+     *
+     * @param uri the uri of the graph must be dropped
+     */
+    void dropGraph(CrawleableUri uri);
+
+    /**
+     * This method updates the graph id of the uriNew ro the graph id of uriOld
+     *
+     * @param uriNew the uri of which the graph id must be updated
+     * @param uriOld the uri with graph id which must be used to update uriNew
+     */
+    void updateGraphForUri(CrawleableUri uriNew, CrawleableUri uriOld);
+
+    /**
+     * This method adds the Graph ids of the uris in the metadata
+     *
+     * @param uris list of uris
+     */
+    void addGraphIdForURIs(List<CrawleableUri> uris);
```

Comment:
Since the methods in SparqlBasedSInk are needed too like fetching triples and closing sink and the dependency need to be removed, is it better to move the common code Squirrel.API project? Creating a class in deduplication will mean not having access to methods like getTriplesForGraph() and closeSinkForUri(). Please let me know your opinion on this.

Reply:
I tried some ways, but they were inefficient. I finally moved the sink part to squirrel.api project and used it in deduplication project. Please have a look at the approach.